ethereum/datastructures/src/test/java/tech/pegasys/artemis/datastructures/operations/AttestationDataTest.java

```diff
@@ -61,12 +64,83 @@ void shouldNotBeProcessableBeforeFirstSlotOfTargetEpoch() {
             UnsignedLong.valueOf(1),
             UnsignedLong.ZERO,
             Bytes32.ZERO,
-            new Checkpoint(UnsignedLong.ONE, Bytes32.ZERO),
+            new Checkpoint(ONE, Bytes32.ZERO),
             target);
 
     assertThat(data.getEarliestSlotForProcessing()).isEqualTo(target.getEpochStartSlot());
   }
 
+  @Test
+  public void shouldNotBeAbleToIncludeInBlockBeforeSlotAfterCreationSlot() {
+    final AttestationData data =
+        new AttestationData(
+            UnsignedLong.valueOf(60),
+            UnsignedLong.ZERO,
+            Bytes32.ZERO,
+            new Checkpoint(ONE, Bytes32.ZERO),
+            new Checkpoint(ONE, Bytes32.ZERO));
+
+    assertThat(data.canIncludeInBlockAtSlot(UnsignedLong.valueOf(60))).isFalse();
+    assertThat(data.canIncludeInBlockAtSlot(UnsignedLong.valueOf(61))).isTrue();
+  }
+
+  @Test
+  public void shouldNotBeAbleToIncludeInBlockBeforeFirstSlotOfTargetEpoch() {
+    final Checkpoint target = new Checkpoint(UnsignedLong.valueOf(10), Bytes32.ZERO);
+    final AttestationData data =
+        new AttestationData(
+            target.getEpochStartSlot().minus(ONE),
```

Comment:
I think the changes necessary for the comment I made above will affect this test as well.